web/src/unread_ops.js

```diff
@@ -46,12 +52,33 @@ export function confirm_mark_all_as_read() {
     confirm_dialog.launch({
         html_heading: $t_html({defaultMessage: "Mark all messages as read?"}),
         html_body,
-        on_click: mark_all_as_read,
+        on_click: mark_stream_topic_or_all_as_read,
         loading_spinner: true,
     });
 }
 
-export function mark_all_as_read(args = {}) {
+function mark_stream_topic_or_all_as_read(args = {}) {
```

Comment:
Why if we just call this `bulk_mark_mesages_as_read`?

Reply:
Also would it be cleaner to just have the callers pass the `narrow` parameter rather than constructing it here?

Reply:
@timabbott Thanks for the review. Updated the code!